PancakeSwap-Bots/Large-swap/src/utils.ts

```diff
@@ -0,0 +1,97 @@
+import { ethers, Finding, FindingType, FindingSeverity } from "forta-agent";
+import BigNumber from "bignumber.js";
+import { getCreate2Address } from "@ethersproject/address";
+import { ERC20ABI, PANCAKE_PAIR_ABI, cache } from "./constants";
+
+BigNumber.set({ DECIMAL_PLACES: 18 });
+
+const toBn = (ethersBn: ethers.BigNumberish) => new BigNumber(ethersBn.toString());
+
+const createContract = (contractAddress: string, abi: string | string[], provider: ethers.providers.Provider) =>
+  new ethers.Contract(contractAddress, abi, provider);
+
+const isValidPancakePair = async (
+  pairAddress: string,
+  provider: ethers.providers.Provider,
+  block: number,
+  pancakeFactoryAddr: string,
+  init: string
+): Promise<[boolean, string, string]> => {
+  const key: string = `pool-${pairAddress}-${block}`;
+  if (cache.has(key)) return cache.get(key) as [boolean, string, string];
+  const pairContract = createContract(pairAddress, PANCAKE_PAIR_ABI, provider);
+  let token0Address: string, token1Address: string;
+  try {
+    [token0Address, token1Address] = await Promise.all([
+      pairContract.token0({ blockTag: block }),
+      pairContract.token1({ blockTag: block }),
+    ]);
+  } catch {
+    return [false, "", ""];
+  }
+  const tokenPair = getPancakePairCreate2Address(pancakeFactoryAddr, token0Address, token1Address, init);
+  const isValid = tokenPair.toLowerCase() === pairAddress.toLowerCase() ? true : false;
+  cache.set(key, [isValid, token0Address, token1Address]);
+  return [isValid, token0Address, token1Address];
+};
+
+const getERC20Balance = async (
```

Comment:
I think for `isValidPancakePair`, and `getERC20Balance` we can create a fetcher file and its accompanying test file.